src/api/polkadot/polkadotContext.ts

```diff
@@ -23,6 +33,36 @@ const mutations = {
     setApi: (apiInst: ApiPromise) => {
         state.api = apiInst;
     },
+    setCurrentAccount: (accountIndex: number) => {
+        const api = state.api;
+        if (api === null) {
+            return;
+        }
+
+        const accounts = keyring.getPairs();
+        const accountsLength = accounts.length;
+        if (accountsLength === 0 || accountIndex > accountsLength || accountIndex < 0) {
+            return;
+        }
+        // unsubscribe previous AccountInfo
+        const unsub = state.unsubscribeAccountInfo;
+        if (unsub) {
+            (async function () {
+                (await unsub)();
+            })();
+        }
+
+        const currentAccount = accounts[accountIndex];
+        state.currentAccount = currentAccount;
+
+        // subscribe current AccountInfo
+        state.unsubscribeAccountInfo = api.query.system.account(
+            currentAccount.address,
+            (result) => {
+                state.currentBalance = result.data.free.toBn();
```

Comment:
I don't think we need to subscribe to the account balance from here. Instead, we can create a separate provider and only use the balance information when the component requires it, rather than loading everything from the start